plugins/user-settings-backend/src/service/router.ts

```diff
@@ -0,0 +1,168 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { PluginDatabaseManager, errorHandler } from '@backstage/backend-common';
+import { AuthenticationError, InputError } from '@backstage/errors';
+import {
+  getBearerTokenFromAuthorizationHeader,
+  IdentityClient,
+} from '@backstage/plugin-auth-node';
+import express, { Request } from 'express';
+import Router from 'express-promise-router';
+
+import { DatabaseUserSettingsStore, UserSettingsStore } from '../database';
+
+/**
+ * @public
+ */
+export async function createUserSettingsStore(database: PluginDatabaseManager) {
+  return await DatabaseUserSettingsStore.create(await database.getClient());
+}
+
+/**
+ * @public
+ */
+export interface RouterOptions<T> {
+  userSettingsStore: UserSettingsStore<T>;
+  identity: IdentityClient;
+}
+
+/**
+ * Create the user settings backend routes.
+ *
+ * @public
+ */
+export async function createRouter<T>(
+  options: RouterOptions<T>,
+): Promise<express.Router> {
+  const { userSettingsStore, identity } = options;
+  const router = Router();
+  router.use(express.json());
+
+  /**
+   * Helper method to extract the userEntityRef from the request.
+   */
+  const getUserEntityRef = async (req: Request): Promise<string> => {
+    const token = getBearerTokenFromAuthorizationHeader(
+      req.header('authorization'),
+    );
+
+    if (!token) {
+      throw new AuthenticationError(`Missing token in 'authorization' header`);
+    }
+
+    // throws an AuthenticationError in case the token is invalid
+    const user = await identity.authenticate(token);
+
+    return user.identity.userEntityRef;
+  };
+
+  // get all user related settings
+  router.get('/', async (req, res) => {
+    const userEntityRef = await getUserEntityRef(req);
+
+    const settings = await userSettingsStore.transaction(tx =>
+      userSettingsStore.getAll(tx, { userEntityRef }),
+    );
+
+    res.json(settings);
+  });
+
+  // remove all user related settings
+  router.delete('/', async (req, res) => {
+    const userEntityRef = await getUserEntityRef(req);
+
+    await userSettingsStore.transaction(tx =>
+      userSettingsStore.deleteAll(tx, { userEntityRef }),
+    );
+
+    res.send(204).end();
+  });
+
+  // get a single bucket
+  router.get('/:bucket', async (req, res) => {
+    const userEntityRef = await getUserEntityRef(req);
+    const { bucket } = req.params;
+
+    const settings = await userSettingsStore.transaction(tx =>
+      userSettingsStore.getBucket(tx, { userEntityRef, bucket }),
+    );
+
+    res.json(settings);
+  });
+
+  // delete a whole bucket
+  router.delete('/:bucket', async (req, res) => {
```

Comment:
deletes should probably also be on the exact same URL patterns as above, so you can get and delete the exact same things